ADR/0032-decoupling-deployment.md

```diff
@@ -0,0 +1,238 @@
+  a high barrier to engage with some of the useful subsystems we've built. A user (and potential
+  contributor) has to adopt the whole thing, or nothing.
+- Today, we can generate simple but often incorrect deployment manifests for users' applications,
+  which we deploy via the application gitops repo. The user cannot interact with or influence these
+  resources directly; they have to do so through the Application/Component API. If we permit them to
+  directly provide deployment resources through that API, the API will become a leaky abstraction.
+  If don't permit them to provide their own resources, and instead try to handle every application
+  configuration case in the HAS generation code, we will struggle to keep up with users' demands for
+  the many and varied kinds of application deployments.
+
+See also [RHTAP-1873](https://issues.redhat.com/browse/RHTAP-1873).
+
+## Decision
+
+We are going to decouple the deployment from the build, test, and release portions of our system.
+
+**Deployment**:
+
+- The [Environment], [SnapshotEnvironmentBinding], and [GitOpsDeploymentManagedEnvironment]
+  resources will be deprecated and eventually decomissioned.
+- The [application-service] will stop generating GitOps repo content and stop creating GitOps
+  repositories.
+- We will stop deploying the [gitops-service] entirely and the RDS instance it uses.
+- HAC will stop rendering [Environments] and their status.
+- If a user wants to make use of deployment capabilities, we will promote the usage of [renovatebot]
+  to propagate _released_ images to their self-managed gitops repo as pull requests. Dependabot is
+  equally viable if the user's gitops repo is on GitHub.
+
+**Test and promotion**:
+
+- We will decomission the [DeploymentTargetClaim], [DeploymentTarget], and [DeploymentTargetClass]
+  APIs in favor of the new [Dynamic Resource Allocation APIs] that are an alpha feature of Kubernetes
+  v1.27 and OpenShift 4.14.
+- [integration-service] should no longer create and and manage [Environments] and the related
+  [DeploymentTargetClaims]. Users will be expected to provide integration test pipelines that (somehow)
+  specify `resourceClaims`, which will cause a provisioner to provision the compute and inject a
+  kubeconfig for the target compute into the taskrun pod. The user's test pipeline should then take
+  steps to *deploy an instance of their application to be tested* to the compute provided by dynamic
+  resource allocation provisioner, using using the provided kubeconfig.
+- In the intervening time between now and when the Dynamic Resource Allocation APIs are available
+  (OpenShift 4.14 plus the time we need to implement a sandbox SpaceRequest *resource driver*), users
+  that need an ephemeral namespace for testing will need to employ a Task as the first step in their
+  pipeline that creates a SpaceRequest. They will need to use a finally task to clean up the SpaceRequest
```

Comment:
In the interim solution, no. I think if the user doesn't clean it up, they will pile up.

In the long-term solution (DRA), then the resource driver should clean it up after the pod is either deleted or terminates.

Reply:
we will set and owner reference on the SpaceRequest (the pipeline run will own it), so it will get deleted when the the PLR will get deleted. In addition we will implement a Chron job to deleted old space requests (just so we will have another safe guard).